src/property.ts

```diff
@@ -10,100 +10,158 @@
 
 'use strict';
 
-const assert = require('assert');
-
-const DESCR_FIELDS = [
-  'title',
-  'type',
-  '@type',
-  'unit',
-  'description',
-  'minimum',
-  'maximum',
-  'enum',
-  'readOnly',
-  'multipleOf',
-  'links',
-];
-function copyDescrFieldsInto(target, source) {
-  // Check 'min' and 'max' for backwards compatibility.
-  if (source.hasOwnProperty('min')) {
-    target.minimum = source.min;
-  }
+import {Device} from './device';
 
-  if (source.hasOwnProperty('max')) {
-    target.maximum = source.max;
-  }
+import assert from 'assert';
 
-  // Check 'label' for backwards compatibility.
-  if (source.hasOwnProperty('label')) {
-    target.title = source.label;
-  }
+interface LegacyPropertyDescription {
+  min: number
+  max: number
+  label: string
+  visible: boolean
+}
 
-  for (const field of DESCR_FIELDS) {
-    if (source.hasOwnProperty(field)) {
-      target[field] = source[field];
-    }
-  }
+export interface PropertyDescription {
+  title: string
+  type: string
+  '@type': string
+  unit: string
+  description: string
+  minimum: number
+  maximum: number
+  enum: string[]
+  readOnly: boolean
+  multipleOf: number
+  links: string[]
 }
 
-class Property {
-  constructor(device, name, propertyDescr) {
+export class Property<T> {
+  public device: Device;
+
+  private name: string;
+
+  private title: string;
+
+  private type: string;
+
+  private '@type': string;
+
+  private unit: string;
+
+  private description: string;
+
+  private minimum: number;
+
+  private maximum: number;
+
+  private enum: string[];
+
+  private readOnly: boolean;
+
+  private multipleOf: number;
+
+  private links: string[];
+
+  private visible = true;
+
+  private fireAndForget = false;
+
+  private value?: T;
+
+  private prevGetValue?: T;
+
+  constructor(device: Device, name: string,
+              propertyDescr: PropertyDescription) {
+    this.device = device;
+
+    this.name = name;
+
     // The propertyDescr argument used to be the 'type' string, so we add an
     // assertion here to notify anybody who has an older plugin.
-
     assert.equal(typeof propertyDescr, 'object',
                  'Please update plugin to use property description.');
 
-    this.device = device;
-    this.name = name;
-    this.visible = true;
-    this.fireAndForget = false;
-    if (propertyDescr.hasOwnProperty('visible')) {
-      this.visible = propertyDescr.visible;
+    const legacyDescription = <LegacyPropertyDescription><unknown>propertyDescr;
+
+    if (legacyDescription.hasOwnProperty('visible')) {
+      this.visible = legacyDescription.visible;
     }
 
-    copyDescrFieldsInto(this, propertyDescr);
+    this.title = propertyDescr.title || legacyDescription.label;
+    this.type = propertyDescr.type;
+    this['@type'] = propertyDescr['@type'];
+    this.unit = propertyDescr.unit;
+    this.description = propertyDescr.description;
+    this.minimum = propertyDescr.minimum || legacyDescription.min;
```

Comment:
I think that is what you want to achieve?

```suggestion
    this.minimum = propertyDescr.minimum ?? legacyDescription.min;
```

See more: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing

Reply:
Good point

Reply:
Done